git/git.go

```diff
@@ -21,23 +21,27 @@ func VerifyRef(ref string) bool {
 
 // CurrentBranch reads the checked-out branch for the git repository
 func CurrentBranch() (string, error) {
-	err := run.PrepareCmd(GitCommand("log")).Run()
-	if err != nil {
-		// this is a hack.
-		errRe := regexp.MustCompile("your current branch '([^']+)' does not have any commits yet")
-		matches := errRe.FindAllStringSubmatch(err.Error(), -1)
-		if len(matches) > 0 && matches[0][1] != "" {
-			return matches[0][1], nil
-		} // else we continue on. it's weird that git log failed but we'll let other errors surface.
-	}
+	noBranchError := errors.New("git: not on any branch")
+
 	// we avoid using `git branch --show-current` for compatibility with git < 2.22
-	branchCmd := exec.Command("git", "rev-parse", "--abbrev-ref", "HEAD")
-	output, err := run.PrepareCmd(branchCmd).Output()
+	output, err := run.PrepareCmd(GitCommand("rev-parse", "--abbrev-ref", "HEAD")).Output()
```

Comment:
This approach looks like it might cover all the cases, but I wonder if we could get away with using just a single invocation without fallbacks? It looks like `git symbolic-ref --quiet --short HEAD` could be just what we need:

* exit code == 0 + lines on stdout: the branch name was determined successfully
* exit code != 0 + no stderr: detached HEAD (no current branch)
* exit code != 0 + lines on stderr: some other error with which we should abort the process

Reply:
great thought, implemented.